File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  attr_reader :numbers
  VERSION = 2

  def initialize(number)
    @numbers = all_numbers(number)
  end

  def all_numbers(number)
    number_array = []
    until number == 0
      number_array << number
      number -= 1
    end
    number_array
  end

  def square_of_sum
    sum = 0
    numbers.each do |number|
      sum += number
    end
    sum ** 2
  end

  def sum_of_squares
    sum = 0
    numbers.each do |number|
      sum += (number ** 2)
    end
    sum
  end

  def difference
    (sum_of_squares - square_of_sum).abs
  end
end

File: php/difference-of-squares/difference-of-squares.php
<?php

function squareOfSums($num)
{
  $result = 0;

  do {
    $result += $num;
    $num -= 1;
  } while ($num > 0);

  return $result**2;
}

function sumOfSquares($num)
{
  $result = 0;

  do {
    $result += $num**2;
    $num -= 1;
  } while ($num > 0);

  return $result;
}

function difference($num)
{
  return squareOfSums($num) - sumOfSquares($num);
}
